### PR TITLE
[WIP] Change coverage to codecov

### PR DESCRIPTION
**Pull request by OriolAbril**
This PR changes code coverage provider from coveralls to codecov to add a check on the code coverage of _only_ newly added lines of code. 

For this to work, codecov must be given access to the repo. This can be done from this page: https://github.com/organizations/pymc-devs/settings/installations/3153839 it looks like only by @twiecki. Once permissions are set, codecov should comment on this PR and add a check on the PR. Both the checks and the comment can be further personalized, we should probably merge this only once we are all happy with the codecov setting and seen it can report back to this PR

Here are a few important guidelines and requirements to check before your PR can be merged:
+ [x] There is an informative high-level description of the changes.
+ [x] The description and/or commit message(s) references the relevant GitHub issue(s).
+ [ ] The commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [ ] The commits correspond to [relevant logical changes](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes).
+ [ ] There are tests covering the changes introduced in the PR.

Don't worry, your PR doesn't need to be in perfect order to submit it.  As development progresses and/or reviewers request changes, you can always [rewrite the history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_rewriting_history) of your feature/PR branches.

If your PR is an ongoing effort and you would like to involve us in the process, simply make it a [draft PR](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests).

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+coverage:
+  status:
+    project:
+      default:
+        target: auto
+        threshold: 0%
+
+    patch:
+      default:
+        target: 100%
```
Comment (twiecki) [inline codecov.yml:18]:
```suggestion
        target: 100%
        
codecov:
  token: 27bb1e3f-0e2c-4c4e-beca-45f213201f9a
```

**Comment by twiecki** [comment]
OK, I added permissions and a token.

File: codecov.yml (hunk 1 above)
Comment (dfm) [reply codecov.yml:18]: We shouldn't need this here. It's better to use the `GITHUB_SECRET` which happens by default the way @OriolAbril has it set up, I think. We would only need this for a private repo.

Comment (OriolAbril) [reply codecov.yml:18]: codecov should not need any token nor secret for public repos like this one, I'll try rerunning the jobs without any of the two and if it does not work add a secret

Comment (dfm) [reply codecov.yml:18]: That's because it's *using* the `GITHUB_SECRET` that is already available by default. No, you shouldn't need to add it manually!

**Comment by twiecki** [comment]
Is it working, can we merge?

**Comment by OriolAbril** [comment]
It looks like every time that a job finishes, codecov edits the comment. This should be fine but it can be confusing to see a coverage of 30% posted in 20 minutes and see how it is progressively increased to the current 70% as the different builds finish. codecov can be configured to wait x jobs before reporting back so this is up to preference.

I also don't know how to change the required from the coveralls job to the 2 codecov ones

**Review by brandonwillard — APPROVED** [review]

**Comment by brandonwillard** [comment]
@OriolAbril, this is no longer a WIP, right?  If not, we can merge this.

**Comment by OriolAbril** [comment]
Yes, this is working already. It can still be customized further, which can be done if desired in follow up PRs

**Comment by brandonwillard** [comment]
> Every time that a job finishes, codecov edits the comment. This should be fine but it can be confusing to see a coverage of 30% posted in 20 minutes and see how it is progressively increased to the current 70% as the different builds finish. codecov can be configured to wait x jobs before reporting back so this is up to preference.

How exactly do we fix this?

**Pull request merged**